src/handlers/psp.ts

```diff
@@ -249,68 +250,48 @@ const initializeProviders = async (state: State): Promise<State> => {
-    const { sponsorAddress, providerState, subscriptions } = subscriptionGroup;
-    const { airnodeWallet, providerName, chainId, provider, contracts, voidSigner, currentBlock, gasTarget } =
-      providerState;
-
-    const providerLogOptions = buildLogOptions('meta', { chainId, providerName }, baseLogOptions);
-
-    // Fetch sponsor wallet transaction counts to be able to assign nonces to subscriptions
-    const [transactionCountLogs, walletData] = await getSponsorWalletAndTransactionCount(
-      airnodeWallet,
-      provider,
-      currentBlock,
-      sponsorAddress
-    );
-
-    // Skip processing for the current sponsorAddress if the wallet functions fail
-    if (isNil(walletData)) {
-      const sponsorLogOptions = buildLogOptions('additional', { sponsor: sponsorAddress }, providerLogOptions);
-      utils.logger.warn('Failed to fetch sponsor wallet or transaction count', sponsorLogOptions);
-      return;
-    }
-
-    const { sponsorWallet, transactionCount } = walletData;
-
-    const sponsorWalletLogOptions = buildLogOptions(
-      'additional',
-      { sponsorWallet: shortenAddress(sponsorWallet.address) },
-      providerLogOptions
-    );
-    utils.logger.logPending(transactionCountLogs, sponsorWalletLogOptions);
-
-    utils.logger.info(`Processing ${subscriptions.length} subscription(s)`, sponsorWalletLogOptions);
-
-    const logs = await processSponsorWallet(
-      airnodeWallet,
-      contracts['DapiServer'],
-      gasTarget,
-      subscriptions,
-      sponsorWallet,
-      voidSigner,
-      transactionCount
-    );
-
-    logs.forEach(([logs, data]) => {
-      const subscriptionLogOptions = buildLogOptions(
-        'additional',
-        { subscriptionId: data.id },
-        sponsorWalletLogOptions
-      );
-      utils.logger.logPending(logs, subscriptionLogOptions);
-    });
-  });
-
-  await Promise.all(providerSponsorPromises);
+export const spawn = ({
+  providerSponsorSubscription,
+  baseLogOptions,
+  type,
+}: {
+  providerSponsorSubscription: ProviderSponsorSubscriptions;
+  baseLogOptions: utils.LogOptions;
+  type: 'local' | 'aws' | 'gcp';
+}) => {
+  // lambda.invoke is synchronous so we need to wrap this in a promise
+  switch (type) {
+    case 'local':
+      return processSubscriptionsHandler({ body: JSON.stringify({ providerSponsorSubscription, baseLogOptions }) });
+    case 'aws':
+      return new Promise((resolve, reject) => {
+        // Uses the current region by default
+        const lambda = new AWS.Lambda();
+
+        // TODO: do we want to use the airnode format for FunctionName
+        // AWS doesn't allow uppercase letters in lambda function names
+        // const resolvedName = `airkeeper-${providerSponsorSubscription.sponsorAddress}-run`;
+
+        const options = {
+          FunctionName: 'process-subscriptions',
+          Payload: JSON.stringify({ providerSponsorSubscription, baseLogOptions }),
```

Comment:
The `airnodeWallet.mnemonic.phrase`  is lost here once this is stringified so I'll update the process-suscriptions lambda to load the config file.

Reply:
The main goal of `initializeProviders` function was to do tasks getting the block number, etc only once. If that goal remains after moving it to `process-subscriptions` then I'm OK with it.